src/wp-includes/media.php

```diff
@@ -5578,7 +5578,45 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 	 * It is here to avoid duplicate logic in many places below, without having
 	 * to introduce a very specific private global function.
 	 */
-	$postprocess = static function( $loading_attributes, $with_fetchpriority = false ) use ( $tag_name, $attr, $context ) {
+	$postprocess = static function( $loading_attributes, $with_fetchpriority = false, $is_header = false ) use ( $tag_name, $attr, $context ) {
+		if ( $is_header ) {
+			/** This filter is documented in wp-admin/includes/media.php */
+			$wp_min_priority_img_pixels = apply_filters( 'wp_min_priority_img_pixels', 50000 );
+			// Images with a certain minimum size in the header of the page are also counted towards the threshold.
+			if ( $wp_min_priority_img_pixels <= $attr['width'] * $attr['height'] ) {
+				wp_increase_content_media_count();
+			}
+		}
+		// Potentially add `fetchpriority="high"`.
+		if ( $with_fetchpriority ) {
+			$loading_attributes = wp_maybe_add_fetchpriority_high_attr( $loading_attributes, $tag_name, $attr );
+		}
+		// Potentially strip `loading="lazy"` if the feature is disabled.
+		if ( isset( $loading_attributes['loading'] ) && ! wp_lazy_loading_enabled( $tag_name, $context ) ) {
+			unset( $loading_attributes['loading'] );
+		}
+		return $loading_attributes;
+	};
+	/*
+	 * Closure for postprocessing logic.
+	 * It is here to avoid duplicate logic in many places below, without having
+	 * to introduce a very specific private global function.
+	 */
```

Comment:
```suggestion
```

None of this is needed, it is just repeated code.

Reply:
@spacedmonkey this was an accidental commit, I have fixed it now. Also, I have moved the counting code to a separate closure.